x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/impacted_spaces_flyout.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, Component } from 'react';
+import PropTypes from 'prop-types';
+import { PrivilegeSpaceTable } from './privilege_space_table';
+import {
+  EuiLink,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiFlyoutBody,
+  EuiTitle,
+} from '@elastic/eui';
+
+import './impacted_spaces_flyout.less';
+import { NO_PRIVILEGE_VALUE } from '../../../lib/constants';
+import { getKibanaPrivilegesViewModel } from '../../../lib/get_application_privileges';
+import { ManageSpacesButton } from '../../../../../../../../spaces/public/components';
+
+export class ImpactedSpacesFlyout extends Component {
+  state = {
+    showImpactedSpaces: false
+  };
+
+  render() {
+    const flyout = this.getFlyout();
+    return (
+      <Fragment>
+        <div className="showImpactedSpaces">
+          <EuiLink onClick={this.toggleShowImpactedSpaces}>See summary of all spaces privileges</EuiLink>
+        </div>
+        {flyout}
+      </Fragment>
+    );
+  }
+
+  toggleShowImpactedSpaces = () => {
+    this.setState({
+      showImpactedSpaces: !this.state.showImpactedSpaces
+    });
+  }
+
+  getFlyout = () => {
+    if (!this.state.showImpactedSpaces) {
+      return null;
+    }
+
+    const {
+      role,
+      spaces,
+      kibanaAppPrivileges,
+      basePrivilege,
+    } = this.props;
+
+    const { assignedPrivileges } = getKibanaPrivilegesViewModel(kibanaAppPrivileges, role.kibana);
+
+    const allSpacePrivileges = spaces.reduce((acc, space) => {
+      const spacePrivilege = assignedPrivileges.space[space.id] ? assignedPrivileges.space[space.id][0] : basePrivilege;
```

Comment:
If the user adds a Space privilege of "read" and then sets the global privilege of "all", we still show the read privilege in the flyout.